### DISCUSSION TITLE
The Ghost Printer Behind Top-level R Expressions | /en/2017/06/top-level-r-expressions/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#937] [Comments]
https://yihui.org/en/2017/06/top-level-r-expressions/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
invisible() is equivalent to invisible(NULL).

An explicit invisible() value at the end of a function will make sure the returned value is invisible. If you rely on other expressions to return an invisible value, you may forget it after you change the function and return a visible value by accident.

> Originally posted on 2017-06-11 04:18:53

**Comment by alanraetz** [comment]
This explanation is very helpful, thank you. I have been stuck for over a day... as a long-term programmer who is new to R and R markdown, it seems a little ridiculous, but once you explain the mechanism, it makes sense. I was assuming it was some bug in the code, I just couldn't isolate it... I know in R loops are not common, but I wish this 'feature' was mentioned more prominently in the documentation.
